speech/infer_darwin.go

```diff
@@ -0,0 +1,110 @@
+//go:build darwin
```

Comment:
Wish there was a clever way to avoid this repetition (e.g. re-mapping missing types only?) but not the end of the world I suppose.

Reply:
There might be a way to split it up even more to get the arch-specific code into one function, I'll look into it.